quest/api/projects.py

```diff
@@ -80,33 +80,14 @@ def new_project(name, display_name=None, description=None, metadata=None,
     if name in projects.keys():
         raise ValueError('Project %s exists, please use a unique name' % name)
 
-    if display_name is None:
-        display_name = name
-
-    if description is None:
-        description = ''
-
-    if folder is None:
-        folder = name
-
-    if metadata is None:
-        metadata = {}
-
-    if not os.path.isabs(folder):
-        path = os.path.join(util.get_projects_dir(), folder)
-    else:
-        path = folder
-
-    util.mkdir_if_doesnt_exist(path)
-    dbpath = os.path.join(path, PROJECT_DB_FILE)
-    db = init_db(dbpath)
-    with db_session:
-        db.Project(display_name=display_name,
-                   description=description,
-                   metadata=metadata)
-    db.disconnect()
-    projects.update({name: {'folder': folder}})
-    _write_projects(projects)
+    _new_project(
```

Comment:
Why did you the creation of the project to another method?

Reply:
When `_load_projects` is called if there are no projects then it needs to be able to to create a default project. Since `new_project` calls  `_load_projects` then I would end up in a recursive loop if it tried to create the default project with `new_project`. I created `_new_project` that handles most of the creation process but doesn't call `_load_projects` so that it can be called from both `new_project` and from `_load_projects`.

Reply:
Okay! Thanks